### ISSUE TITLE
spacy-llm custom task configuration

### ISSUE THREAD

**Issue by aschimmenti** [explosion/spacy-llm#439]
Hi,
Is a more detailed explanation of the custom task configuration in the works? The current documentation is a bit too generic and unclear, and the NoOp file isn't very informative. Any assistance would be greatly appreciated! I'm excited about using this library for other tasks; it fits perfectly with other NLP tasks that don't rely on large language models. Thank you for any help you can provide!

**Comment by svlandeg** [comment]
Hi! Yes, you're right, we need to document this better.

Currently, the best explanation is over at https://github.com/explosion/spacy-llm/tree/main/usage_examples#writing-your-own-task.

Any thoughts on what would be helpful to add to this? Are there particular bits you're stuck on? Does it help looking at the built in tasks we have over at https://github.com/explosion/spacy-llm/tree/main/spacy_llm/tasks? I can imagine that the class inheritance makes it a bit difficult to follow, so maybe we should work on a fully written-out, self-contained example. The recently introduced [sharding](https://spacy.io/api/large-language-models#task-sharding) functionality also makes this slightly more complicated.

**Comment by aschimmenti** [comment]
I'm stuck on multiple points, I get a few sharding configuration errors and a bit of problems in configuring the prompt for few-shot tasks. I could do it with other LLM wrappers but I really enjoy the structured doc output so it can seamlessly go the llm-rel task. The task I'm interested is entity typing, i.e. given an entity, induce the class from the context 
E.g. "Babylonia was an ancient Akkadian-speaking Semitic state and cultural region based in central-southern Mesopotamia (present-day Iraq)."
Output: {"entity": "Babilonia", "class":"state"}, {"entity": "Mesopotamia", "class":"region"}
or
Bājíquán is a Chinese martial art that features explosive, short-range power and is famous for its elbow strikes.
Output: {"entity": "Bājíquán ", "class":"martial art"}

**Comment by svlandeg** [comment]
In our test suite, we have a toy task implemented called `"spacy.CountWithSharding.v1`", you can find it [here](https://github.com/explosion/spacy-llm/blob/main/spacy_llm/tests/sharding/util.py#L46). Maybe you can use that as a "template" for now, while we work on the docs.

With respect to your entity typing task - can't you use Named Entity Recognition for this? `spacy-llm` has built-in tasks for [NER](https://spacy.io/api/large-language-models#ner) as well as [spancat](https://spacy.io/api/large-language-models#spancat) - maybe that will already fulfill your needs for this particular use-case.

**Comment by aschimmenti** [comment]
Hello! Thanks for your suggestion, I managed to use NER.v3 to also recognise classes by providing examples and to connect it to another spacy REL model by combining the pipelines. 
I just have a question: 
The `template` argument in the few-shot NER v3 configuration lets you add a different path for a jinja template. I created a 1:1 jinja template for the NER v3 task just by changing the text, but I always get an empty Entities array. 
In the `fewshot.cfg` file I've wrote: 

`[components.llm.task]  
@llm_tasks = "spacy.NER.v3"  
labels = ["ENTITY", "CLASS"]  
template = "./template-ent-classification.v3.jinja"
`   
Is something wrong there? Thank for any help, it is much appreciated.
